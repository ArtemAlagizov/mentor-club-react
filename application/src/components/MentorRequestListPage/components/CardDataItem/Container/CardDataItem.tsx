import React from "react";
import CardDataItemView from "../View/CardDataItemView";
import { ICardDataItem } from "./ICardDataItem";

const CardDataItem: React.FC<ICardDataItem> = ({ value, label }) => {
  return <CardDataItemView label={label} value={value} />;
};

export default CardDataItem;
