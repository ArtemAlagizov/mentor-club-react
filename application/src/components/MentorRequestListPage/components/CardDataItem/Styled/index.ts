import styled from "styled-components";

const CardDataItem = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  overflow: hidden;

  .card-data-item-text {
    width: 200px;
    text-overflow: ellipsis;
    overflow: hidden;
    text-align: right;
  }
`;

const CardDataItemLabel = styled.div`
  font-weight: bold;
  margin-right: 5px;
`;

const CardDataItemValue = styled.div``;

export const Styled = {
  CardDataItem,
  CardDataItemLabel: CardDataItemLabel,
  CardDataItemValue: CardDataItemValue
};
