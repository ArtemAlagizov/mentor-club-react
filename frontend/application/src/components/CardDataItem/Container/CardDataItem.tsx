import React from "react";
import CardDataItemView from "../View/CardDataItemView";
import { ICardDataItem } from "./ICardDataItem";

const CardDataItem: React.FC<ICardDataItem> = ({ valueContainer, label }) => {
  return <CardDataItemView label={label} valueContainer={valueContainer} />;
};

export default CardDataItem;
