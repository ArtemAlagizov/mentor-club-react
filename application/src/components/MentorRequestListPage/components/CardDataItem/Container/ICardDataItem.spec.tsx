import React from "react";
import { shallow } from "enzyme";
import CardDataItem from "./CardDataItem";

describe("Given a CardDataItem", () => {
  describe("when rendering", () => {
    it("should have CardDataItem matching snapshot", () => {
      const cardDataItem = shallow(<CardDataItem label={""} value={""} />);

      expect(cardDataItem).toMatchSnapshot();
    });
  });
});
