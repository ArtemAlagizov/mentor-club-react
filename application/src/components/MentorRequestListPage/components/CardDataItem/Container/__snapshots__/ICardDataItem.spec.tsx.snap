// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CardDataItem when rendering should have CardDataItem matching snapshot 1`] = `
<CardDataItemView
  label=""
  value=""
/>
`;
